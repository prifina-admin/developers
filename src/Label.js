import React, { forwardRef } from "react";

import styled, { css } from "styled-components";

import { layout, space, typography, color, compose } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";
import PropTypes from "prop-types";

const systemProps = compose(layout, space, typography, color);

const nowrap = props => {
  return props.nowrap
    ? {
        whiteSpace: "nowrap",
      }
    : null;
};
const pointer = props => {
  return props.pointer ? { cursor: "pointer" } : null;
};
const hidden = props => {
  return props.hidden ? { visibility: "hidden" } : null;
};

/*
const VisuallyHidden = styled(Box)`
  border: 0px;
  clip: rect(0px, 0px, 0px, 0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  white-space: nowrap;
  position: absolute;
`;
*/

const LabelTheme = css`
  font-size: ${props => props.theme.textStyles[props.textStyle].fontSize};
  letter-spacing: ${props =>
    props.theme.textStyles[props.textStyle].letterSpacing};
  display: ${props => props.variation};
  width: 100%;
  margin: 0;
  color: ${props => props.theme.textStyles[props.textStyle].color};
  font-weight: ${props => props.theme.textStyles[props.textStyle].fontWeight};
  line-height: ${props => props.theme.textStyles[props.textStyle].lineHeight};
  ${systemProps};
  ${nowrap}
  ${pointer}
  ${hidden}
`;

const LabelElement = styled("label")`
  ${LabelTheme}
`;

const Label = forwardRef((props, ref) => {
  return <LabelElement {...props} ref={ref} />;
});
Label.propTypes = {
  ...styledProps.space,
  ...styledProps.layout,
  ...styledProps.typography,

  variation: PropTypes.oneOf(["block", "inline", "none"]),
  pointer: PropTypes.bool,
};
Label.defaultProps = {
  textStyle: "body",
};

Label.displayName = "Label";
Label.isLabel = true;

export default Label;
