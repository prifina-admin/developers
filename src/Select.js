import React, {forwardRef} from "react";

import styled from "styled-components";
import Button from "./Button";
import {default as styledProps} from "@styled-system/prop-types";
import PropTypes from "prop-types";

const SelectElement = styled.select`
  width:  ${props => props.width || "50%"};
  height: $${props => props.height || "35px"};
  background: ${props => props.backgroundColor || "gray"};
  color: ${props => props.color || "gray"};
  padding-left: ${props => props.paddingLeft || "15px"};
  font-size: ${props => props.fontSize || "14px"};
  border: ${props => props.border};
  margin-left: ${props => props.marginLeft || "10px"};
  border-color: ${props => props.borderColor};
  border-radius: ${props => props.borderRadius};  
    
  option {
    color: ${props => props.optionColor || "black"};
    background: ${props => props.optionBackground || "white"};
    display:  ${props => props.display || "flex"};
    white-space: ${props => props.whiteSpace || "pre"};;
    min-height: ${props => props.minHeight || "20px"};
    padding: ${props => props.padding || "0px 2px 1px"};
  }
`;

export const Select = forwardRef(({headerName, optionNames, ...props }, ref)=> {
    return(
          <SelectElement {...props} ref={ref}>
              <option value="" hidden>
                  {headerName}
              </option>
              {optionNames.map((name, index) => (
                  <option value={index}>{name}</option>
              ))}
          </SelectElement>
  );
});

Select.propTypes = {
    ...styledProps.layout,
    ...styledProps.space,
    ...styledProps.color,
    optionColor: PropTypes.string,
    optionBackground: PropTypes.string,
};

export default Select;