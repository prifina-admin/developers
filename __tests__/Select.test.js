import React, {createRef} from "react";
import renderer from "react-test-renderer";
import ThemeProvider from "../src/theme/ThemeProvider";
import Label from "../src/Label";
import Select from "../src/Select";

test("Link renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Label htmlFor="cabinClass">Cabin Class</Label>
            <Select id="cabinClass" name="cabinClass" defaultValue="Premium Economy">
                <option>Economy</option>
                <option>Premium Economy</option>
                <option>Business</option>
                <option>First Class</option>
                <option>
                    With a super long label that doesn't get clobbered by the chevron
                </option>
            </Select>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

