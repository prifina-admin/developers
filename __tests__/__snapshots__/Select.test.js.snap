// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <label
    className="Label__LabelElement-u5wfo1-0 cOOeVN"
    htmlFor="cabinClass"
  >
    Cabin Class
  </label>
  <div
    className="Box__BoxElement-sc-1iw4lg9-0 gOkXjB Select__StyledBox-sc-1fp3wv9-1 AldyR"
  >
    <select
      className="Select__SelectElement-sc-1fp3wv9-0 jtcsP"
      defaultValue="Premium Economy"
      id="cabinClass"
      name="cabinClass"
    >
      <option>
        Economy
      </option>
      <option>
        Premium Economy
      </option>
      <option>
        Business
      </option>
      <option>
        First Class
      </option>
      <option>
        With a super long label that doesn't get clobbered by the chevron
      </option>
    </select>
    <svg
      className="Icon-y3r77h-0 Select__ClickableIcon-sc-1fp3wv9-2 hDJeva"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M7 10l5 5l5-5H7z\\" fill=\\"#000\\"/>",
        }
      }
      focusable={false}
      height="24"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
</div>
`;
